xbmc/dialogs/GUIDialogFavourites.cpp

```diff
@@ -254,3 +229,40 @@ CFileItemPtr CGUIDialogFavourites::GetCurrentListItem(int offset)
   return (*m_favourites)[item];
 }
 
+bool CGUIDialogFavourites::ChooseAndSetNewName(const CFileItemPtr &item)
+{
+  std::string label(item->GetLabel());
+  if (CGUIKeyboardFactory::ShowAndGetInput(label, CVariant{g_localizeStrings.Get(16008)}, false)) // Enter new title
+  {
+    item->SetLabel(label);
+    return true;
+  }
+  return false;
+}
+
+bool CGUIDialogFavourites::ChooseAndSetNewThumbnail(const CFileItemPtr &item)
+{
+  CFileItemList prefilledItems;
+  if (item->HasArt("thumb"))
+  {
+    const CFileItemPtr current(new CFileItem("thumb://Current", false));
```

Comment:
Would be nice if you use make_shared here. I know we don't do that currently but trying to change that